### PR TITLE
add go global-var test

### PR DESCRIPTION
**Pull request by mullikine**

### CODE REVIEW

File: tenets/codelingo/go/global-var/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      import codelingo/ast/go
+      
+      go.file(depth = any):
+        go.decls:
+          go.value_spec(depth = 1):
+            go.names:
+              @playground.highlight
+              go.ident:
+                private == "true"
+                public == "false"
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/global-var/codelingo.yaml]: These two lines say the same thing, you can remove one.

Comment (BlakeMScurr) [reply tenets/codelingo/go/global-var/codelingo.yaml]: In fact, why not match public globals too?

File: tenets/codelingo/go/global-var/example.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package code
+
+var theVar = 2
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/global-var/example.go]: Might be good to have some test cases for non globals too.

File: tenets/codelingo/go/global-var/codelingo.yaml (hunk 1 above)
Comment (waigani) [reply tenets/codelingo/go/global-var/codelingo.yaml]: Public globals may be used by another package

Comment (mullikine) [reply tenets/codelingo/go/global-var/codelingo.yaml]: Is this a reason for or against allowing public globals?

File: tenets/codelingo/go/global-var/example.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+var myVar1, myVar2 = 1, 2
+
+var _ = 0
+
+// Those are not errors
+var myVar = 1
+
+// Those are fake errors which are currently not detected
+// because they start with 'err' or 'Err' and we don't
+// check if such a variable implements the error interface.
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/global-var/example.go]: I don't see anything in the tenet about checking whether the name of the ident starts with `err` or `Err`. Is this comment accurate?

Comment (mullikine) [reply tenets/codelingo/go/global-var/example.go]: It may need to be split into 2 tenets to cover everything.

Comment (mullikine) [reply tenets/codelingo/go/global-var/example.go]: No idea why those test cases were in there. They did come from the external tool though. I'll make sure I'm getting the right test cases.

Comment (mullikine) [reply tenets/codelingo/go/global-var/example.go]: I see. The owner of the original linter must have re-used some test cases from a different linter. But they are actually testing global variables inside the 'error' code.

**Pull request merged**